docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+    2. `WorkoutSessionAdd` parse the arguments to identify the repetitions and weight for the exercise.
+    3. `WorkoutSessionAdd` calls `WorkOutSession.Storage.writeToFile()` to store information of all exercises recorded. 
+1. Based on `ExecutionResult`, correct response will be printed to user.
+
+All description, warnings and response will be handled by `ui` to ensure consistence across the app.
+The following sequence diagram shows how the add command works
+
+The sequence diagram below summarizes how creating new workout session works:
+![Load Data Sequence Diagram](pictures/jinyang/WorkoutSessionAdd.png)
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.1.2. <a id="deleting-an-exercise">Deleting an Exercise</a>
+
+Users can delete an exercise from a pre-existing list of exercise. The failure to do so will trigger an exception where the user will be notified of 
+the reason, e.g. invalid command or IO related errors. The action will be aborted. If the addition is successful, 
+a new exercise will be added to the exerciselist.
+
+**Implementation**
+
+When the user attempts to create a new workout session, the Ui, WorkoutManagerParser 
+and CommandLib class will be accessed and the following sequence of 
+actions are called to return a command object NewWs.
+
+1. User executes `add benchpress /n 6 /w 120`
+     1. `WorkoutSession` calls `Ui.getUserCommand()` to receive user input.
+     2. `WorkoutSession` calls `WorkoutSessionParser.workoutSessionParser` to convert the input to a string array.
+1. Creation of command object.
+     1. Based on the parsed input, `WorkoutManager` calls `CommandLib` to return the correct Command Object `WorkoutSessionAdd`.
+1. Executing Command
+    1. `WorkoutManager` calls `WorkoutSessionAdd.execute()` with the rest of parsed input.
+    2. `WorkoutSessionAdd` parse the arguments to identify the repetitions and weight for the exercise.
+    3. `WorkoutSessionAdd` calls `WorkOutSession.Storage.writeToFile()` to store information of all exercises recorded. 
+1. Based on `ExecutionResult`, correct response will be printed to user.
+
+All description, warnings and response will be handled by `ui` to ensure consistence across the app.
+The following sequence diagram shows how the add command works
+
+The sequence diagram below summarizes how creating new workout session works:
+![Load Data Sequence Diagram](pictures/jinyang/WorkoutSessionAdd.png)
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.1.3. <a id="listing-all-exercises-in-this-session">Listing All Exercises in This Session</a>
+
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.1.4. Allowing users to view help commands
+
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.1.5. Searching for related exercises
+
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.1.6. Ending the workout session
+
+
+[&#8593; Return to Top](#developer-guide)
+#### 4.4.2. Listing past workout sessions
```

Comment:
![Screenshot 2020-10-29 at 12 49 25 PM](https://user-images.githubusercontent.com/69417796/97526962-30b05600-19e5-11eb-95b3-a2798bc31b22.png)
The return arrow might need to change to dashed line.